src/storage/workspace_storage.rs

```diff
@@ -180,80 +176,86 @@ impl WorkspaceStorage {
+                            .map_err(fs_to_python_error)
+                            .map(|_| py.None())
+                    }))
+                } else {
+                    FutureIntoCoroutine::from(async move {
+                        ws.set_manifest(entry_id.0, manifest, cache_only, removed_ids)
+                            .await
+                            .map_err(fs_to_python_error)
+                    })
+                }
+            }
+            Err(e) => FutureIntoCoroutine::ready(Err(e)),
+        }
     }
 
-    fn set_workspace_manifest(
-        &self,
-        manifest: LocalWorkspaceManifest,
-    ) -> PyResult<FutureIntoCoroutine> {
-        let ws = self.get_storage()?;
+    fn set_workspace_manifest(&self, manifest: LocalWorkspaceManifest) -> FutureIntoCoroutine {
+        let ws = self.get_storage();
 
-        Ok(FutureIntoCoroutine::from(async move {
-            ws.set_workspace_manifest(manifest.0)
+        FutureIntoCoroutine::from(async move {
+            ws?.set_workspace_manifest(manifest.0)
                 .await
                 .map_err(fs_to_python_error)
-        }))
+        })
     }
 
     #[cfg(feature = "test-utils")]
-    fn clear_manifest(&self, entry_id: EntryID) -> PyResult<FutureIntoCoroutine> {
-        let ws = self.get_storage()?;
+    fn clear_manifest(&self, entry_id: EntryID) -> FutureIntoCoroutine {
+        let ws = self.get_storage();
 
-        Ok(FutureIntoCoroutine::from(async move {
-            ws.clear_manifest(&entry_id.0).await.map_err(|e| match e {
+        FutureIntoCoroutine::from(async move {
+            ws?.clear_manifest(&entry_id.0).await.map_err(|e| match e {
                 FSError::LocalMiss(_) => FSLocalMissError::new_err(entry_id),
                 _ => fs_to_python_error(e),
             })
-        }))
+        })
     }
 
     fn create_file_descriptor(&self, manifest: LocalFileManifest) -> PyResult<u32> {
         self.get_storage()
             .map(|ws| ws.create_file_descriptor(manifest.0).0)
     }
 
-    fn load_file_descriptor(&self, py: Python<'_>, fd: u32) -> PyResult<FutureIntoCoroutine> {
-        let ws = self.get_storage()?;
-        let fd = libparsec::types::FileDescriptor(fd);
-
-        let fut = match ws.load_file_descriptor_in_cache(fd) {
-            Ok(manifest) => FutureIntoCoroutine::ready(Ok(LocalFileManifest(manifest).into_py(py))),
-            Err(libparsec::core_fs::FSError::LocalMiss(_)) => {
-                FutureIntoCoroutine::from(async move {
-                    ws.load_file_descriptor(fd)
-                        .await
-                        .map_err(fs_to_python_error)
-                        .map(LocalFileManifest)
-                })
+    fn load_file_descriptor(&self, fd: u32) -> FutureIntoCoroutine {
+        match self.get_storage() {
+            Ok(ws) => {
+                let fd = libparsec::types::FileDescriptor(fd);
+
+                match ws.load_file_descriptor_in_cache(fd) {
```

Comment:
Can you do the same thing by merging the result of `get_storage` & `load_file_descriptor_in_cache` ?